packages/gatekeeper/src/gatekeeper.ts

```diff
@@ -807,12 +1039,16 @@ export default class Gatekeeper {
                 }
 
                 const first = currentEvents[0];
-                const nativeRegistry = first.operation.mdip.registry;
+                const nativeRegistry = first.operation.mdip?.registry;
 
                 if (event.registry === nativeRegistry) {
                     const nextEvent = currentEvents[index + 1];
 
-                    if (nextEvent.registry !== event.registry || compareOrdinals(event.ordinal, nextEvent.ordinal) < 0) {
+                    if (nextEvent.registry !== event.registry ||
+                        compareOrdinals(
```

Comment:
An event ordinal should always be an array of integers. We should not handle the case where it is not, that should be enforced elsewhere.

Reply:
I will update the code so that all instances of ordinal is an array of integers and update the ordinal type accordingly.